uber/site_sections/preregistration.py

```diff
@@ -258,6 +258,8 @@ def register_group_member(self, session, group_id, message='', **params):
 
                 badge_being_claimed = group.floating[0]
                 attendee.registered = badge_being_claimed.registered
```

Comment:
Hmm, this will definitely work for Attendee and Guest badges.

If someone claims a Staff or Supporter badge, this won't work like you'd want, since the new badge will have number 0 instead of whatever number is being claimed.  At this point I don't think we have any unclaimed Staff.Supporter badges but we definitely did at one point.

Maybe just merge/deploy this now, and open a backlog ticket for the preassigned badge case?